nlptest/transform/utils.py

```diff
@@ -4956,140 +7016,140 @@ def create_terminology(ner_data: pd.DataFrame) -> Dict[str, List[str]]:
 
 
 default_label_representation = {'O': 0, 'LOC': 0, 'PER': 0, 'MISC': 0, 'ORG': 0}
-default_ehtnicity_representation = {'black': 0, 'asian':0, 'white':0, 'native_american': 0, 'hispanic': 0, 'inter_racial': 0}
-default_religion_representation = {'muslim': 0, 'hindu':0, 'sikh':0, 'christian':0, 'jain':0, 'buddhist':0, 'parsi':0}
-default_economic_country_representation = {'high_income':0 , 'low_income':0, 'lower_middle_income':0, 'upper_middle_income':0} 
+default_ehtnicity_representation = {'black': 0, 'asian': 0, 'white': 0, 'native_american': 0, 'hispanic': 0,
+                                    'inter_racial': 0}
+default_religion_representation = {'muslim': 0, 'hindu': 0, 'sikh': 0, 'christian': 0, 'jain': 0, 'buddhist': 0,
+                                   'parsi': 0}
+default_economic_country_representation = {'high_income': 0, 'low_income': 0, 'lower_middle_income': 0,
+                                           'upper_middle_income': 0}
 
 
-def get_label_representation_dict(data):
+def get_label_representation_dict(data: List[Sample]) -> Dict[str, int]:
     """
     Args:
         data (List[Sample]): The input data to be evaluated for representation test.
 
     Returns:
         dict: a dictionary containing label representation information.
-    """ 
-    
-    entity_representation={}
+    """
+
+    label_representation = defaultdict(int)
+
     for sample in data:
-        for i in sample.expected_results.predictions:
-            if i.entity=='O':
-                if  i.entity not in entity_representation:
-                    entity_representation[i.entity]=1
-                else:
-                    entity_representation[i.entity]+=1
-                    
-            elif i.entity in ['B-LOC','I-LOC','B-PER','I-PER','B-MISC','I-MISC','B-ORG','I-ORG']:
-                if  i.entity.split("-")[1] not in entity_representation :
-                    entity_representation[i.entity.split("-")[1]]=1
-                else:
-                    entity_representation[i.entity.split("-")[1]]+=1
-                                
+        for prediction in sample.expected_results.predictions:
+            if isinstance(prediction, SequenceLabel):
+                label_representation[prediction.label] += 1
+            elif isinstance(prediction, NERPrediction):
+                if prediction.entity == 'O':
+                    label_representation[prediction.entity] += 1
+                elif prediction.entity in ['B-LOC', 'I-LOC', 'B-PER', 'I-PER', 'B-MISC', 'I-MISC', 'B-ORG', 'I-ORG']:
+                    label_representation[prediction.entity.split("-")[1]] += 1
```

Comment:
NOTE: ideally both `SequenceLabel` and `NERPrediction` would have a `label` attribute, but there is no easy way to handle multiple `Field` aliases with `pydantic`.
This is a temporary fix I will have a deeper look into it when refactoring `Sample`.